<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>TestWidget</class>
 <widget class="QWidget" name="TestWidget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>600</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Test Widget</string>
  </property>
  <layout class="QVBoxLayout" name="mainLayout">
   <property name="spacing">
    <number>20</number>
   </property>
   <item>
    <widget class="QStackedWidget" name="stackedWidget">
     <widget class="QWidget" name="setupPage">
      <layout class="QVBoxLayout" name="setupPageLayout">
       <property name="spacing">
        <number>20</number>
       </property>
       <item>
        <widget class="QLabel" name="titleLabel">
         <property name="font">
          <font>
           <pointsize>24</pointsize>
          </font>
         </property>
         <property name="text">
          <string>Test Setup</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignmentFlag::AlignCenter</set>
         </property>
        </widget>
       </item>
       <item>
        <layout class="QFormLayout" name="formLayout">
         <item row="0" column="0">
          <widget class="QLabel" name="modeLabel">
           <property name="text">
            <string>Test Mode:</string>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QComboBox" name="modeCombo">
           <item>
            <property name="text">
             <string>Word -&gt; Meaning</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>Meaning -&gt; Word</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>Meaning -&gt; Spelling</string>
            </property>
           </item>
           <!-- 新增：Typing测试模式 -->
           <item>
            <property name="text">
             <string>Typing Test (Listen &amp; Type)</string>
            </property>
           </item>
          </widget>
         </item>
         <item row="1" column="0">
          <widget class="QLabel" name="countLabel">
           <property name="text">
            <string>Question Count:</string>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
          <widget class="QSpinBox" name="countSpin">
           <property name="minimum">
            <number>1</number>
           </property>
           <property name="maximum">
            <number>100</number>
           </property>
           <property name="value">
            <number>10</number>
           </property>
          </widget>
         </item>
         <item row="2" column="0">
          <widget class="QLabel" name="timeLabel">
           <property name="text">
            <string>Time Limit (seconds):</string>
           </property>
          </widget>
         </item>
         <item row="2" column="1">
          <widget class="QSpinBox" name="timeSpin">
           <property name="specialValueText">
            <string>No Limit</string>
           </property>
           <property name="minimum">
            <number>0</number>
           </property>
           <property name="maximum">
            <number>600</number>
           </property>
           <property name="value">
            <number>60</number>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="buttonLayout">
         <item>
          <widget class="QPushButton" name="startButton">
           <property name="text">
            <string>Start Test</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="backButton">
           <property name="text">
            <string>Back</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="testPage">
      <layout class="QVBoxLayout" name="testPageLayout">
       <property name="spacing">
        <number>20</number>
       </property>
       <item>
        <layout class="QHBoxLayout" name="topLayout">
         <item>
          <widget class="QLabel" name="timerLabel">
           <property name="text">
            <string>Time left: 60 seconds</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="homeButton">
           <property name="text">
            <string>Home</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <widget class="QLabel" name="questionLabel">
         <property name="font">
          <font>
           <pointsize>20</pointsize>
          </font>
         </property>
         <property name="text">
          <string>Question</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignmentFlag::AlignCenter</set>
         </property>
        </widget>
       </item>
       <item>
        <layout class="QVBoxLayout" name="optionsLayout">
         <property name="spacing">
          <number>10</number>
         </property>
        </layout>
       </item>
       <item>
        <widget class="QLineEdit" name="spellingInput">
         <property name="visible">
          <bool>false</bool>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="submitButton">
         <property name="visible">
          <bool>false</bool>
         </property>
         <property name="text">
          <string>Submit</string>
         </property>
        </widget>
       </item>
       <item>
        <spacer name="verticalSpacer">
         <property name="orientation">
          <enum>Qt::Orientation::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
       <item>
        <widget class="QLabel" name="progressLabel">
         <property name="text">
          <string>Question 1/10</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="resultsPage">
      <layout class="QVBoxLayout" name="resultsPageLayout">
       <property name="spacing">
        <number>30</number>
       </property>
       <item>
        <widget class="QLabel" name="resultsTitleLabel">
         <property name="font">
          <font>
           <pointsize>24</pointsize>
          </font>
         </property>
         <property name="text">
          <string>Test Results</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignmentFlag::AlignCenter</set>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="scoreLabel">
         <property name="font">
          <font>
           <pointsize>20</pointsize>
          </font>
         </property>
         <property name="text">
          <string>Score: 0/0</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignmentFlag::AlignCenter</set>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="percentLabel">
         <property name="text">
          <string>Percentage: 0.0%</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignmentFlag::AlignCenter</set>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="timeUsedLabel">
         <property name="text">
          <string>Time used: 0 seconds</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignmentFlag::AlignCenter</set>
         </property>
        </widget>
       </item>
       <item>
        <layout class="QHBoxLayout" name="resultsButtonLayout">
         <item>
          <widget class="QPushButton" name="restartButton">
           <property name="text">
            <string>New Test</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="resultsBackButton">
           <property name="text">
            <string>Back to Word List</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <widget class="QTableView" name="resultTableView">
          <property name="sizePolicy">
            <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
            </sizepolicy>
          </property>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections/>
</ui>
